gcp_variant_transforms/options/variant_transform_options.py

```diff
@@ -438,9 +438,13 @@ def add_arguments(self, parser):
     parser.add_argument(
         '--output_file',
         required=True,
-        help='The full path of the VCF file to store the result.')
+        help='The path of the VCF file to store the result inside the bucket.')
     parser.add_argument(
         '--input_table',
         required=True,
         help=('BigQuery table that will be loaded to VCF. It must be in the '
               'format of (PROJECT:DATASET.TABLE).'))
+    parser.add_argument(
```

Comment:
I think this API is a bit confusing; what do you think of using the `--temp_location` flag (just dynamically create a new folder under there), store all of the intermediate VCF files there, and then combine them into the output_file. That way, the user ends up with a single file under output_file, and it would be atomic (i.e. the file would only be copied if all intermediate ones were successful). This is actually the idea that BQ I/O uses as well.

Reply:
Thanks! My concern is that if using `--temp_location`, then even for the local runner this flag is required, which is not the case right now. 

By providing the `bucket_name`, all intermediate files are stored here (we can create an temp file). It is also atomic since the final VCF file will be formed by composing the meta-information file, data header file, and one composed VCF file (not implemented yet).

Another reason I replaced the full path of the output_file with `blob_name`(I didn't change the flag to blob_name though) and `bucket_name` is that it seems that there is no parser to parse a gs link (e.g., gs://bucket_name/blob_name), but we need those for composing purposes.

Reply:
For DirectRunner: you can just create a tempfolder; we already do this for merging headers (see https://github.com/googlegenomics/gcp-variant-transforms/blob/master/gcp_variant_transforms/vcf_to_bq.py#L169). You do raise an important point though that for DirectRunner, we shouldn't use GCS compose and probably safe to set the bucket size to max value (i.e. just one per chromosome) as the expectation is that you're running with small amount of data. Note that DirectRunner support is important as (depending on how efficient it is), we would use it for htsget and there shouldn't be any dependency on GCS.

For gs parser: I suspect you wouldn't need it if you just use the `--temp_location` flag as you can create that yourself?

Reply:
Thanks. I removed the `bucket_name` flag as suggested. I happened to find a gcs path parser in `gcsio`, which solves the parsing problem.
FYI, the [gsutil ](https://github.com/GoogleCloudPlatform/gsutil) also supports parsing and provides various cloud storage APIs. But we would end up with dependency compatibility issues if using it. To be more specific, `apache-beam` requires `google-apitools >=0.5.18, < 0.5.20`, where `gsutil 4.28` depends on `google-apitools ==0.5.16` and `gsutil 4.29` requires `google-apitools == 0.5.21`.